### DISCUSSION TITLE
Modernize recharts.org website

### DISCUSSION THREAD

**Discussion by akamfoad** [recharts/recharts.org#269] [Ideas]
Greetings, 

I have noticed [Recharts Website 1.0](https://github.com/orgs/recharts/projects/5) project and I couldn't help but notice that, the website doesn't in need of some migrations like testing or TS migration, but also under the hood. Looks like its the case not just with website, but also for Recharts the library, and other libraries in this organization, which all use custom Webpack configurations, that are genius but complex and slow to be honest on a decent machine.

I'd like to get your ideas about this because I think having a modern tooling not just boost DX for maintainers but also helps other contributors to be able to swiftly spin up the project, change it and see the results.

The website, I think can take good advantage of modern frameworks such as Next.js or Remix.js, this way we kinda ignore the tooling under the hood and can focus on the Website, its UX, correctness, SEO and more.

For example with Remix, we could build a page once and use HTTP headers to cache it for quite sometime as they don't change that much, which improves UX a lot, we can take https://pagespeed.web.dev as a baseline to improve the website.

But even with no fancy framework, there is still modern build toolings for Client Rendered apps, such as [Vite](https://vitejs.dev/) and [Redux Toolkit](https://redux-toolkit.js.org/) to replace custom Redux boilerplate, which is the recommended way of building apps for the new projects that use Redux.

I can say the same about the libraries, even though they're much complicated, but I think migrating them to take advantage of modern tools such as [SWC](https://swc.rs), [ESBuild](https://github.com/evanw/esbuild) and etc, which I think would boost the performance and iteration a lot.

I'd love to hear what you have about these points, and I can expand more on each one. I'm hoping to bring much more life to this project.

**Comment by ckifer** [comment]
Hey, thanks for bringing this up. 

I would say this would be a good step, but we're trending toward moving away from the website and entirely and onto storybook as the platform for the docs. We're going to try to theme the storybook to look a bit like the current recharts site. Overall it should be a lot easier to maintain the docs in a centralized location such as the storybook.

**Reply by akamfoad** [reply]
Makes sense, so in the future we'd only have the Storybook served on recharts website?

**Reply by ckifer** [reply]
Yep it would be the storybook on the recharts domain

**Reply by akamfoad** [reply]
Sounds good

**Comment by ckifer** [comment]
In terms of the other repositories, the only one that gets much use is react-smooth which needs refactored to TS at least for for sure. Moving to other build tools would be very nice, but we can't guarantee 100% we won't break consumers when we do if we produce different output than we currently do.

**Reply by akamfoad** [reply]
I think with a major version bump and it'll be fine for the react-smooth, even though I just casually started this, it needs investigation to see how it has been utilized and what are the popular ways of downloading this, unpkg or npm? etc..

But what about Recharts, I know its soon to get to this but I think it pretty much needs some modernization too, so to speak.

**Reply by ckifer** [reply]
react-smooth is mostly consumed via recharts itself. A MV bump of react-smooth warrants a MV bump of recharts if it's actually breaking. 

Recharts needs lots of love 😅, but the consumers expect it to be packaged the way that it currently is. Maybe we can plan to change build tool to vite or something? But I would want to wait for next MV to do it

**Reply by akamfoad** [reply]
Indeed, I was shooting these out of curiosity, also there was already a v3 board so it might be possible to have that major version, but most probably not very soon😁
